client/src/mock/api/expense/expenses.ts

```diff
@@ -1,67 +1,109 @@
 import { response } from '../_common/axios_response';
 
-import expenses from '$mock/entities/user_expense.json';
+import expensesJson from '$mock/entities/user_expense.json';
 import recurrenceTypes from '$mock/entities/recurrence_type.json';
 import currencyTypes from '$mock/entities/currency_type.json';
 import transactionsJson from '$mock/entities/transaction.json';
-
+import predefinedExpenses from '$mock/entities/predefined_expense.json';
+import type { Expense, Transaction } from '$shared/domain';
 import type {
 	DeleteTransactionRequestData,
 	DeleteTransactionResponse,
-	Expense,
-	GetCurrencyTypesResponse,
-	GetExpensesRequestData,
-	GetExpensesResponse,
-	NewTransactionRequestData,
-	NewTransactionResponse,
-	Transaction,
-} from '../models/expense_model';
+} from '$shared/api/deleteTransaction';
+import type { GetCurrencyTypesResponse } from '$shared/api/getCurrencyTypes';
+import type { GetPredefinedExpensesResponse } from '$shared/api/getPredifinedExpenses';
+import type { GetRecurrenceTypesResponse } from '$shared/api/getRecurrenceTypes';
+import type { NewExpenseRequestData, NewExpenseResponse } from '$shared/api/newExpense';
+import type { NewTransactionRequestData, NewTransactionResponse } from '$shared/api/newTransaction';
+import type { GetExpensesRequestData, GetExpensesResponse } from '../../../shared/api/getExpenses';
 
+let expenses: Expense[] | null = null;
 let transactions: Transaction[] | null = null;
 
+export const mockGetPredefinedExpenses = (): GetPredefinedExpensesResponse => {
+	const data = predefinedExpenses.map((expense) => {
+		const recurrenceType = recurrenceTypes.find(
+			(recurrenceType) => recurrenceType.id === expense.recurrence_type_id
+		);
+		const currencyType = currencyTypes.find(
+			(currencyType) => currencyType.id === expense.currency_type_id
+		);
+		return {
+			id: expense.id,
+			name: expense.name,
+			description: expense.description,
+			recurrenceType: recurrenceType ? recurrenceType : { id: 0, name: '', perYear: 0 },
+			currencyType: currencyType ? currencyType : { id: 0, abbreviation: '', name: '' },
+			value: expense.value,
+		};
```

Comment:
transactions declaration is missing here (with value being an []). probably this is allright.

Reply:
What do you mean?